src/table/src/engine.rs

```diff
@@ -104,7 +106,19 @@ pub trait TableEngine: Send + Sync {
     /// Drops the given table. Return true if the table is dropped, or false if the table doesn't exist.
     async fn drop_table(&self, ctx: &EngineContext, request: DropTableRequest) -> Result<bool>;
 
-    /// Close the table.
+    /// Closes the (partial) given table.
+    ///
+    /// Removes a table if all regions are closed.
+    /// Returns true if table is removed in the engine.
+    /// Returns false if table is only partial closed.
```

Comment:
How about returns the number of closed region?

Reply:
However, if we return to the closed regions, it makes things get tricky. It's better to know whether the table was removed from the engine. Then, we can deregister it in the catalog.